vms/rpcchainvm/ghttp/http_test.go

```diff
@@ -84,6 +86,58 @@ func TestRequestClientArbitrarilyLongBody(t *testing.T) {
 	client.ServeHTTP(w, r) // Shouldn't block forever reading the body
 }
 
+// Tests that writes to the http response in the server are propagated to the
+// client
+func TestHttpRequest(t *testing.T) {
+	require := require.New(t)
+
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(123)
+		w.Header()["foo"] = []string{"bar"}
+		_, err := w.Write([]byte{1, 2, 3})
+		require.NoError(err)
+	})
+
+	listener, err := grpcutils.NewListener()
+	require.NoError(err)
+	server := grpc.NewServer()
+	httppb.RegisterHTTPServer(server, NewServer(handler))
+
+	go func() {
+		require.NoError(server.Serve(listener))
+	}()
+
+	conn, err := grpc.NewClient(
+		listener.Addr().String(),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	require.NoError(err)
+
+	client := NewClient(httppb.NewHTTPClient(conn))
+
+	recorder := &httptest.ResponseRecorder{
+		Body: bytes.NewBuffer(nil),
+	}
+
+	request := &http.Request{
+		Body: io.NopCloser(strings.NewReader("bar")),
+		Header: http.Header{
+			"Upgrade": {"upgrade"},
```

Comment:
We have to specify this header because this fixes the bug when we're not using http simple. I'm also wondering if we should just remove `handleHttpSimple` entirely...